File: src/main/java/com/ByTech/ByTech/Cart/repositories/DetailRepository.java
package com.ByTech.ByTech.Cart.repositories;

import com.ByTech.ByTech.Cart.models.Detail;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import java.util.List;

public interface DetailRepository extends JpaRepository<Detail, Long> {
    List<Detail> findBySale_Id(Long saleId);

    @Query("SELECT SUM(s.amount) FROM Sale s WHERE FUNCTION('MONTH', s.date) = :month AND s.fabricante = :enterpriseId")
    Long findByMonthAndEnterpriseId(@Param("month") int month, @Param("enterpriseId") Long amount);
}

File: Movil/backend/ByTech_movil/src/main/kotlin/com/example/ByTech_movil/Detail/repositories/DetailRepository.kt
package com.example.ByTech_movil.Detail.repositories

import com.example.ByTech_movil.Detail.models.Detail
import org.springframework.data.jpa.repository.JpaRepository

interface DetailRepository : JpaRepository<Detail, Long> {
    fun findBySale_Id(saleId: Long?): List<Detail>
}
